tasks/src/main/java/error_exceptions/service/StudentsContainer.java

```diff
@@ -0,0 +1,134 @@
+        String firstName) {
+        for (Student student : students) {
+            if (student.getLastName().equals(lastName) && student.getFirstName()
+                .equals(firstName)) {
+                averageMark = averageMark(student);
+            }
+        }
+        if (averageMark == 0.0) {
+            throw new StudentSubjectsException("Student " + lastName + " " + firstName
+                + " without marks or not study in this university");
+        }
+        return averageMark;
+    }
+
+    private static boolean studentInGroup(List<Faculty> faculties, Group.GroupName group) {
+        boolean hasStudent = false;
+        for (Faculty faculty : faculties) {
+            String groupNames = faculty.getGroupName().name();
+            if (groupNames.contains(group.name())) {
+                hasStudent = true;
+                break;
+            }
+        }
+        return hasStudent;
+    }
+
+    private static boolean groupsInFaculty(List<Faculty> faculties, Group.GroupName group,
+        Faculty.FacultyName facultyName) {
+        boolean hasGroup = false;
+        for (Faculty faculty : faculties) {
+            if (faculty.getFacultyName().equals(facultyName) && faculty.getGroupName()
+                .equals(group)) {
+                hasGroup = true;
+                break;
+            }
+        }
+        return hasGroup;
+    }
+
+    private static boolean facultyInUniversity(List<Faculty> faculties,
+        Faculty.FacultyName facultyName) {
+        boolean hasFaculty = false;
+        for (Faculty faculty : faculties) {
+            String facultyNames = faculty.getFacultyName().name();
+            if (facultyNames.contains(facultyName.name())) {
+                hasFaculty = true;
+                break;
+            }
+        }
+        return hasFaculty;
+    }
+
+    public static double averageMarkOfSpecificSubjectGroupFaculty(List<Faculty> faculties,
+        List<Student> students, Student.SubjectsName subject,
+        Group.GroupName group, Faculty.FacultyName facultyName) {
+        sumAllMarks = 0;
+        countSubjects = 0;
+        averageMark = 0.0;
+
+        if (!studentInGroup(faculties, group)) {
+            throw new StudentInGroupException("Group without student");
+        }
+
+        if (!groupsInFaculty(faculties, group, facultyName)) {
+            throw new GroupsInFacultyException("Faculty not contains this group");
+        }
+
+        if (!facultyInUniversity(faculties, facultyName)) {
+            throw new FacultyInUniversityException("Faculty not exist");
+        }
+
+        for (Faculty faculty : faculties) {
+            for (Student student : students) {
+                if (faculty.getStudent().getLastName().equals(student.getLastName()) &&
+                    faculty.getStudent().getFirstName().equals(student.getFirstName()) &&
+                    faculty.getGroupName() == group && faculty.getFacultyName() == facultyName
```

Comment:
faculty.getGroupName().equals(group)
faculty.getFacultyName().equals(facultyName)
You don't need to compare links to objects, so use equals check